docs/scos/dev/feature-walkthroughs/202204.0/configurable-product-feature-walkthrough/configurable-product-flow-chart.md

```diff
@@ -0,0 +1,198 @@
+
+### Phase 4
+
+1. The request comes from the gateway to the configurator `index.php` (phase 4.1).
+2. The configurator handles the request by creating a new session and saving the request's data there (phase 4.2).
+3. The configurator responds with a redirect URL that contains what session ID needs to be picked up (phase 4.2).
+4. The request comes from the gateway to the configurator `index.php` to pick up a session (phase 4.3).
+5. Redirects the customer to the configurator page by a GET request using a secured connection.
+
+### Phase 5
+
+1. You are on the configurator page.
+2. The configurator page is rendered based on the data in the session.
+3. The customer does the configuration.
+4. Submits the configuration form.
+5. The configurator redirects the customer to the gateway page with configuration data.
+
+| PARAMETER | VALUE | COMMENT |
+|---|---|---|
+| `ProductConfigurationInstance.price` | `123` | Sensitive data. |
+| `ProductConfigurationInstance.isComplete` | `true` | Sensitive data. |
+| `ProductConfigurationInstance.availableQuantity` | `2` |   |
+| `ProductConfigurationInstance.displayDescription` | Some text of JSON blob—for example, `{"color"=>"red", "weight"=> 100}` |   |
+| `ProductConfigurationInstance.configuration` | `{"color"=>"red", "weight"=> 100}`   | Sensitive data. |
+| `idCustomer` | `DE-1` |   |
+| `sourceType` | `pdp`, `cart-page` |   |
+| `SKU`  | `some_sku` |   |
+| `timestamp` | `1231313123123` |   |
+| `CheckSum` | It’s an encrypted value of the `CheckSum`. It should be based on the all requested parameters and should have the same order for decryption. |   |
+
+### Phase 6
+
+1. The user finishes the configuration and clicks a button which creates an AJAX POST request with the data to the configurator page (self). (phase 6.1)
+2. In the backend, the response is prepared according to the public data API from the configurator to Spryker. (phase 6.2)
+3. In the backend, a `CheckSum` is prepared based on the response data, which is encrypted with a shared key and returns these as the AJAX response (phase 6.3)
+4. The data is put to a hidden form on the configurator page and submit the form that points to the gateway page. (phase 6.4)
+5. After the successful configuration, the customer is redirected to the configurator gateway with a configuration response.
+6. The gateway URL does not equal to the back URL; it’s a fixed, known URL.
+
+### Phase 7
+
+1. The gateway page receives data.
+2. The data is checked through the execution of the validator plugins stack for received data.
+  - If validation is not successful, the request is redirected to the `backUrl` without saving the of the configuration with a warning message.
+  - If the validation part is successful, the configuration is saved to the session.
+4. All applicable plugins that can handle the configurator response are executed.
+  - A plugin that applies for the PDP source type resolves the back URL according to the response data.
```

Comment:
Is it a substep or we should make it step 5?
```suggestion
5. A plugin that applies for the PDP source type resolves the back URL according to the response data.
```

Reply:
It's part of the step I think. *One* of the plugins that are executed does that.